### ISSUE TITLE
Library search : not the same results in advanced search and in favorites

### ISSUE THREAD

**Issue by liowalter** [swissbib/vufind#679]
If I search etoy in the [library field in the advanced search](https://www.swissbib.ch/Search/Advanced), I don't find any library.

If I search etoy in the [favorite library search](https://www.swissbib.ch/MyResearch/Favorites) then I find the "Mediathèque d'Etoy".

The advanced search only search in the label. The tokenizing is not strict enough. I need to search "d'etoy" to find the lib. For St-Prex, I need to search St-Prex. Prex doesn't work

The favorite search is more fault-tolerant. It also searches in the addresses, the library codes and other fields.

**Comment by liowalter** [comment]
Also : basel hauptbibliothek gives nothing

**Comment by liowalter** [comment]
![](https://github.trello.services/images/mini-trello-icon.png) [Library search : not the same results in advanced search and in favorites](https://trello.com/c/jsaEnnTq/410-library-search-not-the-same-results-in-advanced-search-and-in-favorites)

**Comment by edelm** [comment]
see also https://github.com/swissbib/vufind/issues/31

**Comment by liowalter** [comment]
Favorite Search : 

- work using https://github.com/swissbib/vufind/blob/master/themes/sbvfrd/js/swissbib/FavoriteInstitutions.js based on https://github.com/swissbib/vufind/blob/master/themes/bootstrap3/js/lib/autocomplete.js (from vufind)
- uses [labels + adresses](https://github.com/swissbib/vufind/blob/master/module/Swissbib/src/Swissbib/Libadmin/Importer.php#L489) from local storage (favorite-institutions.ini). If language is changed there is a need to remove cache. But it searches in the user's language
- on mobile it offers a search as well
- the order is important (uni basel works, but basel uni doesn't work)
- it search any part of a word (usanne will find lausanne)

Advanced search
- uses this [jquery plugin](https://harvesthq.github.io/chosen/) for autocompletion
- uses the label only, in the user language
- basically it is a big select
- on mobile it only shows a list
- the order is important (uni basel works, but basel uni doesn't work)
- it searches at the beginning of a word (usanne doesn't find lausanne). But this can be [configured](https://harvesthq.github.io/chosen/options.html).

**Comment by liowalter** [comment]
The above commits solves the problem with `Mediathèque d'Etoy` and `Bibliothèque communale de St-Prex`. Search Etoy and Prex will find this both libraries.

To offer the next feature (searching UB Basel or Basel UB should deliver the same results), neither [chosen jquery plugin](https://harvesthq.github.io/chosen/) nor [autocomplete.js](https://github.com/swissbib/vufind/blob/master/themes/bootstrap3/js/lib/autocomplete.js offers this option. Even the static version of autocomplete.js doesn't offer this.

We need another tool. Possible candidates : 
- javascript libraries like [Lunr](https://lunrjs.com/guides/getting_started.html)
- create a new index in SOLR and search there
- use the mysql fulltext search as it is currently working in [libadmin](https://admin.swissbib.ch/libadmin/)
- do a search in elastic in the linked index (the institutions should be there thanks to this morph : https://github.com/linked-swissbib/mfWorkflows/blob/master/transformation/indexWorkflows/libadminMorph.xml)
- do a search via data.swissbib.ch (https://data.swissbib.ch/organisation/NEBIS-Z01 used to work in the past).

Or we let it as it is...
